builder/builder_test.go

```diff
@@ -179,17 +179,21 @@ func TestBuild(t *testing.T) {
 				eventDataTx := getEventData[bfttypes.EventDataTx](t, subscription)
 				checkTxResult(t, &eventDataTx.TxResult, wantBlock, i, tx)
 
-				wantBlockEvents = append(wantBlockEvents, eventDataTx.Result.Events...)
+				expectedTxResults = append(expectedTxResults, &eventDataTx.TxResult.Result)
 			}
 
+			var expectedBlockEvents []abcitypes.Event
 			eventDataNewBlockEvents := getEventData[bfttypes.EventDataNewBlockEvents](t, subscription)
 			require.Equal(t, int64(wantBlock.Header.Height), eventDataNewBlockEvents.Height)
 			require.Equal(t, int64(len(wantBlock.Txs)), eventDataNewBlockEvents.NumTxs)
-			require.Equal(t, wantBlockEvents, eventDataNewBlockEvents.Events)
+			require.Equal(t, expectedBlockEvents, eventDataNewBlockEvents.Events)
 
 			eventDataNewBlock := getEventData[bfttypes.EventDataNewBlock](t, subscription)
 			require.Equal(t, wantBlock.ToCometLikeBlock(), eventDataNewBlock.Block)
 			require.Equal(t, bytes.HexBytes(wantBlock.Header.Hash.Bytes()), eventDataNewBlock.BlockID.Hash)
+			require.Equal(t, expectedBlockEvents, eventDataNewBlock.ResultFinalizeBlock.Events)
+			require.Equal(t, expectedTxResults, eventDataNewBlock.ResultFinalizeBlock.TxResults)
+			require.Equal(t, postBuildInfo.GetLastBlockAppHash(), eventDataNewBlock.ResultFinalizeBlock.AppHash)
```

Comment:
Can we assert that the `bfttypes.EventDataNewBlock` struct is equal in one go, rather that testing equality on each of its fields? This way, if new fields are added later, the test will fail (as it should).

I didn't catch this in the first review.

Reply:
Good idea, I added this in a new commit